File: packages/engine/src/cpus/6502/instructions/inc.test.js
import { test, expect } from 'vitest'
import { BufferMapper } from '../../../buffer_mapper'
import { CPU } from '..'

test('inc', () => {
  const mmu = new BufferMapper()
  const cpu = new CPU(mmu)
  mmu.data = [0xe6, 0x2, 0xff] // INC, $02
  const g = cpu.run()
  for (let i = 0; i < 5; i++) g.next()
  expect(mmu.data[0x2]).toBe(0)
})

File: packages/engine/src/cpus/6502/instructions/inc.test.zig
const testing = @import("../testing/setup.zig");

test "INC" {
    var cpu = testing.createCPU();
    cpu.write(0, 0xe6);
    cpu.write(1, 2);
    cpu.write(2, 0xff);
    for (0..5) |_| {
        cpu.process();
    }
    try testing.expectEqual(0, cpu.read(2));
}
